executor/table_reader.go

```diff
@@ -177,6 +181,9 @@ func (e *TableReaderExecutor) Open(ctx context.Context) error {
 // Next fills data into the chunk passed by its caller.
 // The task was actually done by tableReaderHandler.
 func (e *TableReaderExecutor) Next(ctx context.Context, req *chunk.Chunk) error {
+	if e.sampler != nil {
```

Comment:
Doesn't consider the generated column?

Reply:
`sampler` uses the KV interface so the generated columns are handled in `DecodeAndEvalRowWithMap`.